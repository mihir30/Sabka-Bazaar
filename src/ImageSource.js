import { Image } from "./components/atoms";
const ImageSource = ({ alt, source, className }) => {
  return (
    <Image source={require(`${source}`)} alt={alt} className={className} />
  );
};

export default ImageSource;
