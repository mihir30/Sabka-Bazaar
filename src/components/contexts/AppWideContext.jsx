import React, { createContext, useReducer } from "react";
import { reducer } from "../reducers/ReducerContext";

const initialState = {
  count: 0,
  products: {},
};

export const AppWideContext = createContext(initialState);

const AppWideContextProvider = ({ children }) => {
  const [cartItems, dispatch] = useReducer(reducer, initialState);

  return (
    <AppWideContext.Provider value={{ cartItems, dispatch }}>
      {children}
    </AppWideContext.Provider>
  );
};

export default AppWideContextProvider;
