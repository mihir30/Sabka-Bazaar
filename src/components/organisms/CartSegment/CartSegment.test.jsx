import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { AppWideContext } from '../../contexts/AppWideContext';
import renderer from 'react-test-renderer';
import ImageSource from '../../../ImageSource';
describe("Cart Segment Component", () => {
    it('renders the item image source component properly', () => {

        render(<AppWideContext.Provider value={{
            cartItems: {
                count: 1, products: {
                    id: "5b6c6a7f01a7c38429530883", imageUrl: "./static/images/products/fruit-n-veg/kiwi-green.jpg",
                    name: "Fresho Kiwi - Green, 3 pcs", price: 87, stock: 50,
                    quantity: 1,
                }
            }
        }}><BrowserRouter><ImageSource alt={"Image of Fresho Kiwi - Green, 3 pcs"} className="image-size-1" source={"./static/images/products/fruit-n-veg/kiwi-green.jpg"} /></BrowserRouter></AppWideContext.Provider>);
        screen.getByAltText("Image of Fresho Kiwi - Green, 3 pcs")


    });


    it('renders the lowest price image source component properly', () => {

        render(<AppWideContext.Provider value={{
            cartItems: {
                count: 1, products: {
                    id: "5b6c6a7f01a7c38429530883", imageUrl: "./static/images/products/fruit-n-veg/kiwi-green.jpg",
                    name: "Fresho Kiwi - Green, 3 pcs", price: 87, stock: 50,
                    quantity: 1,
                }
            }
        }}><BrowserRouter><ImageSource alt="lowest-price" source={"./static/images/lowest-price.png"} className="image-size-2" /></BrowserRouter></AppWideContext.Provider>);
        screen.getByAltText("lowest-price")


    });


    it('should have the right cart item image component snapshot', () => {
        const tree = renderer.create(<AppWideContext.Provider value={{
            cartItems: {
                count: 1, products: {
                    id: "5b6c6a7f01a7c38429530883", imageUrl: "./static/images/products/fruit-n-veg/kiwi-green.jpg",
                    name: "Fresho Kiwi - Green, 3 pcs", price: 87, stock: 50,
                    quantity: 1,
                }
            }
        }}><BrowserRouter><ImageSource alt={"Image of Fresho Kiwi - Green, 3 pcs"} className="image-size-1" source={"./static/images/products/fruit-n-veg/kiwi-green.jpg"} /></BrowserRouter></AppWideContext.Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should have the right lowest price image component snapshot', () => {
        const tree = renderer.create(<AppWideContext.Provider value={{
            cartItems: {
                count: 1, products: {
                    id: "5b6c6a7f01a7c38429530883", imageUrl: "./static/images/products/fruit-n-veg/kiwi-green.jpg",
                    name: "Fresho Kiwi - Green, 3 pcs", price: 87, stock: 50,
                    quantity: 1,
                }
            }
        }}><BrowserRouter><ImageSource alt="lowest-price" source={"./static/images/lowest-price.png"} className="image-size-2" /></BrowserRouter></AppWideContext.Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });


});