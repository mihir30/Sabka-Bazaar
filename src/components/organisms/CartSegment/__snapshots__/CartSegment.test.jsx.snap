// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Segment Component should have the right cart item image component snapshot 1`] = `
<img
  alt="Image of Fresho Kiwi - Green, 3 pcs"
  className="image-size-1"
  data-testid="image"
  src="kiwi-green.jpg"
/>
`;

exports[`Cart Segment Component should have the right lowest price image component snapshot 1`] = `
<img
  alt="lowest-price"
  className="image-size-2"
  data-testid="image"
  src="lowest-price.png"
/>
`;
